scripts/ci-entrypoint.sh

```diff
@@ -132,91 +132,104 @@ create_cluster() {
 }
 
 install_addons() {
-    # Get cluster CIDRs from Cluster object
-    CIDR0=$(${KUBECTL} get cluster "${CLUSTER_NAME}" -o=jsonpath='{.spec.clusterNetwork.pods.cidrBlocks[0]}')
-    CIDR1=$(${KUBECTL} get cluster "${CLUSTER_NAME}" -o=jsonpath='{.spec.clusterNetwork.pods.cidrBlocks[1]}' 2> /dev/null) || true
-
+    CIDRS_RETRIEVED="false"
+    CLUSTER_READY="false"
+    CALICO_SYSTEM_KUBEADM_CONFIG_CREATED="false"
+    CALICO_INSTALLED="false"
+    CIDR0=""
+    CIDR1=""
+    while [[ "${CIDRS_RETRIEVED}" != "true" ]]; do
+        # Get cluster CIDRs from Cluster object
+        CIDR0=$(${KUBECTL} get cluster "${CLUSTER_NAME}" -o=jsonpath='{.spec.clusterNetwork.pods.cidrBlocks[0]}') || continue
+        if [[ $(${KUBECTL} get cluster "${CLUSTER_NAME}" -o=jsonpath='{.spec.clusterNetwork.pods.cidrBlocks}' | jq '. | length') == 2 ]]; then
+            CIDR1=$(${KUBECTL} get cluster "${CLUSTER_NAME}" -o=jsonpath='{.spec.clusterNetwork.pods.cidrBlocks[1]}') || continue
+        fi
+        CIDRS_RETRIEVED="true"
+    done
     # export the target cluster KUBECONFIG if not already set
     export KUBECONFIG="${KUBECONFIG:-${PWD}/kubeconfig}"
+    while [[ "${CLUSTER_READY}" != "true" ]]; do
+        # Copy the kubeadm configmap to the calico-system namespace. This is a workaround needed for the calico-node-windows daemonset to be able to run in the calico-system namespace.
+        cat <<EOF | "${KUBECTL}" apply -f - || continue
```

Comment:
Is this more robust somehow than `kubectl create ns calico-system`?

Reply:
it's idempotent so we don't have to worry about keeping track of its success separately and then not running it during subsequent retry cycles

Reply:
`kubectl create namespace <add-namespace-here> --dry-run=client -o yaml | kubectl apply -f -` is equivalent and slightly easier on the eyes

Reply:
nice!

Reply:
done